packages/snap/src/index.ts

```diff
@@ -161,9 +159,28 @@ async function getPrivateKey(bip44Node?: BIP44Node, index: number = 0): Promise<
  * @returns - The address of the optimal feeCurrency, or undefined if the optimal 
  * feeCurrency is Celo. 
  */
-async function getOptimalFeeCurrency(tx: SimpleTransaction): Promise<string | undefined> {
-  // TODO
-  return undefined
+async function getOptimalFeeCurrency(tx: SimpleTransaction, wallet: CeloWallet): Promise<string | undefined> {
+  const gasLimit = (await wallet.estimateGas(tx)).mul(5)
+  const celoBalance = await wallet.getBalance();
+  const addresses = [ //get this dynamically based on network.
+    "0x874069Fa1Eb16D44d622F2e0Ca25eeA172369bC1",
+    "0x10c892A6EC43a53E45D0B916B4b7D383B1b78C0F",
+    "0xE4D517785D091D3c54818832dB6094bcc2744545"
+  ]; 
```

Comment:
this could be done with the Celo Registry Contract 

something kinda like

```
import RegistryAbi from '@celo/abis/Registry.json'

RegistryContract = new Contract("000...Ce10", RegistryAbi, wallet)

const address = RegistryContract.getAddress('StableToken')


```


```

Reply:
Great tip! That's what's covered by this ticket @montera82 https://github.com/celo-org/celo-monorepo/issues/10328

Reply:
> Great tip! That's what's covered by this ticket @montera82 [celo-org/celo-monorepo#10328](https://github.com/celo-org/celo-monorepo/issues/10328)

While the above seemed very elegant, i lost time trying to implement it cos of the unexplanatory error i got. So skipped that for now to come back to it
https://github.com/celo-org/gas-snap/pull/16#discussion_r1226665404